api/src/main/java/org/pmiops/workbench/db/dao/UserServiceImpl.java

```diff
@@ -445,21 +447,54 @@ private DbUser updateDuccAgreement(
     return dbUser;
   }
 
+  /**
+   * This method is used by create Account flow, it throws exception if user has not accepted AoU
+   * Terms of Service at all or has accepted the incorrect version
+   *
+   * @param tosVersion
+   */
   @Override
-  public void validateTermsOfService(Integer tosVersion) {
+  public void validateAllOfUsTermsOfService(Integer tosVersion) {
     if (tosVersion == null) {
       throw new BadRequestException("Terms of Service version is NULL");
     }
     if (tosVersion != LATEST_AOU_TOS_VERSION) {
-      throw new BadRequestException("Terms of Service version is not up to date");
+      throw new BadRequestException("All of Us Terms of Service version is not up to date");
     }
   }
 
+  // Returns true if user has accepted the latest AoU Terms of Service Version
   @Override
-  public void validateTermsOfService(@Nonnull DbUser dbUser) {
-    final int tosVersion =
-        userTermsOfServiceDao.findByUserIdOrThrow(dbUser.getUserId()).getTosVersion();
-    validateTermsOfService(tosVersion);
+  public boolean validateAllOfUsTermsOfServiceVersion(@Nonnull DbUser dbUser) {
+    try {
+      final int tosVersion =
+          userTermsOfServiceDao.findByUserIdOrThrow(dbUser.getUserId()).getTosVersion();
+      return tosVersion == LATEST_AOU_TOS_VERSION;
+    } catch (BadRequestException ex) {
+      // In case user doesnt have any terms of service
+      return false;
+    }
```

Comment:
We don't need to throw an exception and catch it here.  This also works:
```suggestion
    return userTermsOfServiceDao
        .findFirstByUserIdOrderByTosVersionDesc(dbUser.getUserId())
        .map(u -> u.getTosVersion() == LATEST_AOU_TOS_VERSION)
        .orElse(false);
```